skcosmo/pcovr/kpcovr.py

```diff
@@ -0,0 +1,383 @@
+            scaled so that each feature has a variance of 1 / n_features.
+
+        Y: array-like, shape (n_samples, n_properties)
+            Training data, where n_samples is the number of samples and
+            n_properties is the number of properties
+
+            It is suggested that :math:`\\mathbf{X}` be centered by its column-
+            means and scaled. If features are related, the matrix should be scaled
+            to have unit variance, otherwise :math:`\\mathbf{Y}` should be
+            scaled so that each feature has a variance of 1 / n_features.
+
+        Yhat: array-like, shape (n_samples, n_properties), optional
+            Regressed training data, where n_samples is the number of samples and
+            n_properties is the number of properties. If not supplied, computed
+            by ridge regression.
+
+        Returns
+        -------
+        self: object
+            Returns the instance itself.
+
+        """
+
+        X, Y = check_X_y(X, Y, y_numeric=True, multi_output=True)
+        self.X_fit_ = X.copy()
+
+        if self.n_components is None:
+            self.n_components = min(X.shape)
+
+        self._centerer = KernelFlexibleCenterer()
+        K = self._get_kernel(X)
+
+        if W is None:
+            if Yhat is None:
+                W = (np.linalg.pinv(K, rcond=self.regularization_) @ Y).reshape(
+                    X.shape[0], -1
+                )
+            else:
+                W = np.linalg.lstsq(K, Yhat, rcond=self.regularization_)[0]
+
+        if Yhat is None:
+            Yhat = K @ W
+
+        self._fit(K, Yhat, W)
+
+        self.ptk_ = self.pt__ @ K
+        self.pty_ = self.pt__ @ Y
+
+        if self.fit_inverse_transform:
+            self.ptx_ = self.pt__ @ X
+
+        self.pky_ = self.pkt_ @ self.pty_
+
+        self.components_ = self.pkt_.T  # for sklearn compatibility
+        return self
+
+    def predict(self, X=None, T=None):
+        """Predicts the property values"""
+
+        check_is_fitted(self, ["pky_", "pty_"])
+
+        if X is None and T is None:
+            raise ValueError("Either X or T must be supplied.")
+
+        if X is not None:
+            X = check_array(X)
+            K = self._get_kernel(X)
+            return K @ self.pky_
+        else:
+            T = check_array(T)
+            return T @ self.pty_
+
+    def transform(self, X):
+        """
+        Apply dimensionality reduction to X.
+
+        X is projected on the first principal components as determined by the
+        modified Kernel PCovR distances.
+
+        Parameters
+        ----------
+        X: array-like, shape (n_samples, n_features)
+            New data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        """
+
+        check_is_fitted(self, ["pkt_", "X_fit_"])
+
+        X = check_array(X)
+        K = self._get_kernel(X, self.X_fit_)
+        return K @ self.pkt_
+
+    def inverse_transform(self, T):
```

Comment:
Do we also want an inverse transform to the kernel using `self.ptk_`?

Reply:
sklearn KPCA does "inverse transform" back to X, despite the caveats, so I figured we would do the same here.